### ISSUE TITLE
Ctrl-based shortcuts don't work on Linux

### ISSUE THREAD

**Issue by jacobmischka** [zeit/hyper#274]
In the menus, most shortcuts are listed using the `ctrl` key as the only modifier (likely an automatic artifact from macOS's `cmd`). Some examples:
- Copy `ctrl+c`
- Paste `ctrl+v`
- Close tab `ctrl+w`
- New tab `ctrl+t`

These shortcuts don't work, nor should they since they are essential shortcuts used within the shell. Copy and paste seem to work with the usual `ctrl+shift+c/v`, but I couldn't find working versions of any others in my few minutes of guessing.

**Comment by marcuslindfeldt** [comment]
The default shortcuts for new tabs on Linux is `Ctrl+Shift+T` to open a new tab and `Ctrl+Shift+W` to close one. Currently `Ctrl+w` closes the entire app.

But the reason behind it not working is because of hterm binding all keyboard shortcuts for terminal input and probably preventsDefault() so the keyboard event is not bubbling down to electron. See `app/lib/hterm.js:30`

**Comment by nloomans** [comment]
The problem is that the hterm things that when the meta key is pressed it's for hyperterm, and when not for the shell.
```js
// passthrough all the commands that are meant to control
// hyperterm and not the terminal itself
const oldKeyDown = hterm.Keyboard.prototype.onKeyDown_;
hterm.Keyboard.prototype.onKeyDown_ = function (e) {
  if (e.metaKey) {
    return;
  }
  return oldKeyDown.call(this, e);
};
```

It should check if the shortcut is in use by hyperterm and if not send it to the shell.

**Comment by jacobmischka** [comment]
Agreed, and then also the menu entries need to be updated to show the correct key combinations.

**Comment by willwhitney** [comment]
Related: 
In macOS you can set keyboard shortcuts (in the system preferences) that will trigger a menu item by name. If you set next/previous tab to ctrl+tab/ctrl+shift+tab via this method, they still get captured by hyperterm, and it inserts escape characters.

**Comment by nloomans** [comment]
I already fixed it, but my internet is to shitty to create a pull request... stay tuned.

**Comment by jaydson** [comment]
On my Ubuntu 14.04:

Ctrl+Shift+N > Not working
Ctrl+Shift+T > Not working
Ctrl+Shift+W > Not working
Ctrl+Shift+Z > Not working
Ctrl+Shift+X > Doesn't cut, copy instead
Ctrl+Shift+Z > OK
Ctrl+Shift+V > OK
Ctrl+Shift+A > Not working
Ctrl+Shift+K > Kind working, it clears all the terminal (looks like a bug), but Ctrl+L is working just fine
Ctrl+R > Doesn't reload, it starts the reverse search instead (which is the right thing on Linux)
Ctrl+R > Same as Ctrl+R
Ctrl+Shift+I > Not working
Ctrl+0 > OK
Zoom in and Zoom out are working with Ctrl++ and Ctrl+-
Ctrl+Shift+U > Not working
Ctrl+Alt+Left > Not working
Ctrl+Alt+Right > Not working
F11 > Not working

**Comment by arthurflachs** [comment]
Did anyone come with an idea on how to fix this ?

Apparently keyboard shortcuts are registered through the menu accelerators. Wouldn't it be better to register them through a specific service which would manage all shortcuts inside Hyperterm ?

**Comment by arthurflachs** [comment]
Most "Cmd" shortcuts (corresponding to the windows key) are also not working on Gnome 3 (with Fedora 24). They are used to manage windows position, and the events never reach Hyperterm.
"Command" shortcuts are not natural on Linux anyway.

**Comment by marcuslindfeldt** [comment]
Im with @arthurflachs that it would be great to centralize all keyboard shortcuts in some sort of service. Maybe something like https://github.com/atom/atom-keymap that could map a keyboard shortcut to an action.

**Comment by jdormit** [comment]
@nloomans what was your fix for this?

**Comment by jaydson** [comment]
I don't think it was implemented yet @jdormit , but i'm curious to see what @nloomans did :)

**Comment by jdormit** [comment]
Ah, I see. I am as will! I'm pretty new to the HyperTerm project, but if I find a work-around or a solution I'll post it here (and maybe do a pull request, if it is a good solution lol)

**Comment by cjb** [comment]
@rauchg Hi, any preference on which fix for this is the right one?  Would you like to see https://github.com/atom/atom-keymap used?

**Comment by rauchg** [comment]
@cjb that seems like a really good solution. Like @nloomans pointed out, our approach is really naive at the moment (equating meta key to app and otherwise to terminal).

We're using mousetrap to capture shortcuts in addition to the ones set up on [menubar accelerators](https://github.com/zeit/hyperterm/blob/master/app/menu.js#L22):
https://github.com/zeit/hyperterm/blob/7a01ec77c4edcec506b79a2f8ec10cee633849cd/lib/containers/hyperterm.js#L48

**Comment by toroidal-code** [comment]
So... I don't know if this has changed or not, or if this is a solution, but @jaydson's list of shortcuts all worked perfectly with the addition of the super key. e.g. `super + ctrl + shift + i` to open the dev console, `super + ctrl + shift + u`  to update all of the plugins, etc. Even `super + F11` to go fullscreen, oddly enough.

**Comment by CrashyBang** [comment]
@toroidal-code on Linux?

**Comment by toroidal-code** [comment]
Yeah. Well, I just tried out the text selection short and *those* aren't working. Which is just weird. But might be because my text-selection bindings are globally set to Emacs on my machine.

**Comment by RasCarlito** [comment]
I'm using ubuntu 14.04 and I just have to add the `super` key to every shortcut listed in the menus.
For example, open a new tab is `ctrl + t`, on ubuntu `super + ctrl + t`.

Would be nice if I didn't have to add that `super` though :+1:

**Comment by CrashyBang** [comment]
@RasCarlito Using Ubuntu 16.04 can confirm that adding `super` to the above key combo's fixes them!

**Comment by synthecypher** [comment]
Can you expose the keybindings in `.hyperterm.js` so we can quickly override them there?

**Comment by rumpelsepp** [comment]
<kbd>super</kbd> is not on option for me, since I am using this as a meta key in i3. :/

**Comment by fearphage** [comment]
> Can you expose the keybindings in .hyperterm.js so we can quickly override them there?

I'd prefer that they work correctly out of the box.

**Comment by bogas04** [comment]
Please add https://github.com/zeit/hyper/issues/274#issuecomment-241699565 as a workaround in some docs until this gets fixed.

**Comment by aleksijohansson** [comment]
<kbd>super</kbd> not working for me on Arch Linux with Gnome 3 and I'm having all the same issues described in https://github.com/zeit/hyper/issues/274#issuecomment-235455944

**Comment by matheuss** [comment]
Yo everyone please test #1058! 🙌

**Comment by n370** [comment]
So what's the actual way to workaround? I'm having no luck trying to use screen because the <kbd>ctrl</kbd> + <kbd>a</kbd> on hyper binds to select all text visible and not to screen's default control characters. I'm working on a Fedora Workstation with hyper 1.1.0. I appreciate any comment.

**Comment by cjb** [comment]
Same, ctrl-a and ctrl-r are no longer available to bash, they get used for Hyper bindings instead. This presumably wasn't the intention? I suppose one way forward would be to disable bindings like ctrl-a that conflict with common uses?

The usual thing on Linux is for the terminal to take ctrl-shift-n, not ctrl-n.

**Comment by flybayer** [comment]
@cjb, @n370 it looks like #1191 will hopefully fix this once and for all!

**Comment by divyanshu013** [comment]
Hey guys, reverse search doesn't seems to work. Any workaround for that?

**Comment by ppot** [comment]
#1509

**Comment by rapha8l** [comment]
Hi,

Any news ?
Thanks

**Comment by ppot** [comment]
CLOSED in #1876

**Comment by rapha8l** [comment]
Thanks !
Any idea for the date of a future release ?
